File: standardize-images.mjs

import sharp from 'sharp';
import fs from 'fs';
import path from 'path';

const baseDir = './assets';
const targetSize = 1000;

async function walk(dir) {
    let files = fs.readdirSync(dir);
    for (const file of files) {
        let fullPath = path.join(dir, file);
        if (fs.lstatSync(fullPath).isDirectory()) {
            if (!file.toLowerCase().endsWith('_360')) {
                await walk(fullPath);
            }
        } else if (['.webp', '.jpg', '.png', '.jpeg'].includes(path.extname(file).toLowerCase())) {
            await processImage(fullPath);
        }
    }
}

async function processImage(filePath) {
    try {
        const buffer = fs.readFileSync(filePath);
        const metadata = await sharp(buffer).metadata();

        if (metadata.width === targetSize && metadata.height === targetSize) return;

        console.log(`⏳ A processar: ${filePath} (${metadata.width}x${metadata.height})`);

        const tempPath = filePath + '.tmp';
        await sharp(buffer)
            .resize(targetSize, targetSize, {
                fit: 'cover',
                position: 'center'
            })
            .webp({ quality: 85 })
            .toFile(tempPath);

        fs.renameSync(tempPath, filePath);
        console.log(`✅ Concluído: ${filePath}`);
    } catch (e) {
        console.error(`❌ Erro em ${filePath}: ${e.message}`);
    }
}

console.log("🚀 Iniciando padronização 1000x1000 (Ignorando 360)...");
walk(baseDir);

File: standardize_images.py

import os
from PIL import Image

def process_images(target_size=1000):
    assets_path = './assets'
    valid_extensions = ('.webp', '.jpg', '.png', '.jpeg')
    
    print(f"🚀 Iniciando padronização para {target_size}x{target_size}...")

    for root, dirs, files in os.walk(assets_path):
        if '_360' in root.lower():
            continue

        for file in files:
            if file.lower().endswith(valid_extensions):
                file_path = os.path.join(root, file)
                try:
                    with Image.open(file_path) as img:
                        width, height = img.size
                        
                        if width == target_size and height == target_size:
                            continue

                        ratio = max(target_size / width, target_size / height)
                        new_size = (int(width * ratio), int(height * ratio))
                        
                        img = img.resize(new_size, Image.Resampling.LANCZOS)
                        
                        left = (img.width - target_size) / 2
                        top = (img.height - target_size) / 2
                        right = (img.width + target_size) / 2
                        bottom = (img.height + target_size) / 2
                        
                        img = img.crop((left, top, right, bottom))
                        
                        if file.lower().endswith('.webp'):
                            img.save(file_path, 'WEBP', quality=85)
                        else:
                            img.save(file_path)
                            
                        print(f"✅ Processado: {file_path}")
                except Exception as e:
                    print(f"❌ Erro em {file_path}: {e}")

if __name__ == "__main__":
    process_images()
